src/test/resources/s3-valid-outbound-advice-cases.xml

```diff
@@ -0,0 +1,42 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<beans xmlns="http://www.springframework.org/schema/beans"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:integration="http://www.springframework.org/schema/integration"
+	xmlns:int-aws="http://www.springframework.org/schema/integration/aws"
+	xsi:schemaLocation=
+	"http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd
+	http://www.springframework.org/schema/integration/aws http://www.springframework.org/schema/integration/aws/spring-integration-aws.xsd
+	http://www.springframework.org/schema/integration http://www.springframework.org/schema/integration/spring-integration.xsd">
+
+	<integration:channel id="input"/>
+
+	<int-aws:s3-outbound-channel-adapter
+			id="withHandlerChain"
+			channel="input"
+			bucket="TestBucket"
+			remote-directory="/"
+			charset="US-ASCII"
+			thread-pool-executor="executor"
+			multipart-upload-threshold="5120"
+			temporary-suffix=".write"
+			file-name-generation-expression="headers['name']" propertiesFile="awscredentials.properties">
+		<int-aws:request-handler-advice-chain>
+			<bean class="org.springframework.integration.aws.s3.config.xml.AmazonS3OutboundChannelAdapterParserAWSTests$FooAdvice"/>
+		</int-aws:request-handler-advice-chain>
+	</int-aws:s3-outbound-channel-adapter>
+
+	<bean id="queue" class="java.util.concurrent.ArrayBlockingQueue">
+		<constructor-arg value="10"/>
+	</bean>
+
+	<bean id="executor" class="java.util.concurrent.ThreadPoolExecutor">
+		<constructor-arg index="0" value="5"/>
+		<constructor-arg index="1" value="10"/>
+		<constructor-arg index="2" value="10"/>
+		<constructor-arg index="3" value="MILLISECONDS"/>
+		<constructor-arg index="4" ref="queue"/>
+	</bean>
```

Comment:
Would you mind explaining the reason of this custom `ThreadPoolExecutor`?
Why isn't  `<task:executor>` just enough?

Oh! I see this is just copy/paste from existing tests.

Please, simplify the test config. Just because it doesn't relate to the test case and I don't see reason to pollute it with redundant code.